rs/https_outcalls/client/src/client.rs

```diff
@@ -116,6 +122,7 @@ impl NonBlockingChannel<CanisterHttpRequest> for CanisterHttpAdapterClientImpl {
         let query_handler = self.query_service.clone();
         let metrics = self.metrics.clone();
         let subnet_type = self.subnet_type;
+        let delegation_from_nns = self.delegation_from_nns.get().cloned();
```

Comment:
Unfortunately, there is no `wait` method for `OnceCell` so we either return an error if the cell has not been initialized yet or we don't use the certificate for the time the cert has not been set (this can happen just when starting up).